lib/galaxy/webapps/galaxy/api/uploads.py

```diff
@@ -0,0 +1,54 @@
+"""
+API operations for uploaded files in storage.
+"""
+import logging
+import os
+import re
+
+from galaxy.exceptions import MessageException, NotImplemented
+from galaxy.web import expose_api_anonymous
+from galaxy.web.base.controller import BaseAPIController
+
+log = logging.getLogger(__name__)
+
+
+class UploadsAPIController(BaseAPIController):
+
+    READ_CHUNK_SIZE = 2 ** 16
+    BYTES_PER_MEGABYTE = 1048576
+
+    @expose_api_anonymous
+    def index(self, trans, **kwd):
+        raise NotImplemented("Listing uploads is not implemented.")
+
+    @expose_api_anonymous
+    def create(self, trans, payload, **kwd):
+        """
+        POST /api/uploads/
+        """
+        session_id = payload.get("session_id")
+        session_start = payload.get("session_start")
+        session_chunk = payload.get("session_chunk")
+        if re.match('^[\w-]+$', session_id) is None:
+            raise MessageException("Requires a session id.")
+        if session_start is None:
+            raise MessageException("Requires a session start.")
+        if not hasattr(session_chunk, "file"):
+            raise MessageException("Requires a session chunk.")
+        target_file = os.path.join(trans.app.config.new_file_path, session_id)
+        target_size = 0
+        if os.path.exists(target_file):
+            target_size = os.path.getsize(target_file)
+        if session_start != target_size:
+            raise MessageException("Incorrect session start.")
+        chunk_size = os.fstat(session_chunk.file.fileno()).st_size / self.BYTES_PER_MEGABYTE
```

Comment:
I think you need to add `from __future__ import division` at the top of this file, otherwise you get the floor of the division result in Python2.